llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -71,32 +72,29 @@ def get_run_name(training_run_name: str, current_interval: str) -> str:
     return f'{RUN_NAME_PREFIX}-{current_interval}-{name_without_uuid_suffix}'
 
 
-def get_latest_checkpoint(event: Event, state: State) -> Optional[str]:
-    """Get the latest checkpoint from the training run.
+SUPPORTED_UNITS = {TimeUnit.EPOCH, TimeUnit.BATCH}
+
+
+def get_interval_from_checkpoint(checkpoint: str, unit: TimeUnit) -> Time:
```

Comment:
This and some of the other path wrangling I'm doing feels sort of hacky. I couldn't find anything obvious in composer that I could reuse, please point me to suggestions if you have any

Reply:
yeah this is pretty painful and fragile. What if we saved more info on the `CheckpointSaver` callback in composer instead of trying to pull stuff out of strings here?

Reply:
Here's my proposition: https://www.notion.so/checkpoint-saver-state-improvements-423c8b7f35ca40a0baee678ef63bf8c6

I know we're pretty aggressive with locking to the latest composer in foundry, but what would you think about a hackier string parsing approach until the next composer release / composer bump in foundry? Even if a little more brittle, I think this logic makes the async callback significantly more usable. When manually testing, the previous version was consistently missing intermediate checkpoints